src/server.rs

```diff
@@ -331,3 +346,66 @@ async fn run_server(
     }
     Ok(())
 }
+
+struct RedisServer {
```

Comment:
Is there a phobia of being afraid to see too many lines? (separating Redis to another file?